tensorflow/core/kernels/mkl/mkl_maxpooling_op.cc

```diff
@@ -336,7 +342,8 @@ class MklMaxPoolingGradOp : public MklPoolingBackwardOpBase<T> {
           pooling_bwd->GetPoolingBwdPd();
       T* diff_dst_data = nullptr;
       if (IS_DIFF_DST_REORDER_NEEDED(diff_dst_md, pooling_bwd_pd,
-                                     pooling_bwd)) {
+                                     pooling_bwd) &&
+          !this->native_format_) {
```

Comment:
Nit: Let's check the bool first since it's quicker.
```c++
      if (!this->native_format_ &&
          IS_DIFF_DST_REORDER_NEEDED(diff_dst_md, pooling_bwd_pd,
                                     pooling_bwd)) {	   
```

Reply:
Done.